### PR TITLE
Add LibMeshSystemVectors

### PR DESCRIPTION
**Pull request by drwells**
Part of #932, but applied to IBFEMethod first. @boyceg what do you think?

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by boyceg** [comment]
Looks good to me. What do you think about possibly making the accessor function names a bit more succinct?

**Comment by drwells** [comment]
I like `IBGhostedVectors` since that's the same name we use in `FEDataManager` - it makes it clear that they do the same thing.

I also find `getVector` a bit wordy. I thought about also adding `getVec` to return a PETSc vector immediately but it looks a little confusing to me since it could also just be shorthand for `getVector` (and we only save three characters). Ultimately I'd rather it be a little longer and clear because I can't think of good shorthand that's not overly obscure (we could also abuse `operator()` or `operator[]` but thats going to make things harder to read).

**Comment by boyceg** [comment]
Sounds good.

Are you able to go ahead and implement this change more completely in `IBFEMethod`? I actually have some code that I have been dragging my heels to write because of the painful way that these vectors are currently set up. :-D

**Comment by drwells** [comment]
> Are you able to go ahead and implement this change more completely in IBFEMethod?

Yes. I wanted to post the in-progress version in case someone sees something else we could do to make things easier. I'll convert the rest of the systems (the nice thing about this class is we can do it one system at a time).

File: ibtk/include/ibtk/LibMeshSystemVectors.h [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+     * Get a specific vector for a specific part.
+     */
+    libMesh::PetscVector<double>& getVector(const std::string& vec_name, const unsigned int part)
+    {
+        TBOX_ASSERT(part < d_fe_data_managers.size());
+        if (!vec_stored_in_map(vec_name))
+        {
+            if (vec_name == "solution") return dynamic_cast<libMesh::PetscVector<double>&>(*d_systems[part]->solution);
+            if (vec_name == "current local solution")
+                return dynamic_cast<libMesh::PetscVector<double>&>(*d_systems[part]->current_local_solution);
```
Comment (drwells) [inline ibtk/include/ibtk/LibMeshSystemVectors.h]: Note that we hard-code some special vector names here since libMesh doesn't put these in its map of vectors - `system->solution` is not the same as `system->request_vector("solution")` but for convenience we are slightly inconsistent with libMesh here and identify every vector with a string.

**Comment by drwells** [comment]
On second thought - `vecs->getVectors` is a little redundant. I suppose we could just write  `vecs->get` and `vecs->getIBGhosted` instead.

**Comment by boyceg** [comment]
> On second thought - `vecs->getVectors` is a little redundant. I suppose we could just write `vecs->get` and `vecs->getIBGhosted` instead.

That is exactly what I was thinking when I asked. :-) But this relies on this particular variable naming convention, which does not seem ideal.

**Comment by drwells** [comment]
well, the point of this class is to be used internally, so that's not so bad.

**Comment by boyceg** [comment]
OK, in that case, I would just use `get`.

Also, can we use `"current_local_solution"` (or make it a synonym)?

**Comment by drwells** [comment]
Good idea. I also made `"current"` an alias for `"current_local_solution"`.

**Comment by drwells** [comment]
OK, that should do it.

We shouldn't convert the rest of the systems right now because we have zero tests (and also zero examples) for fluid sources and stress normalization - changing these might break something I don't understand so we'll skip it.

Notably, I did not add any tests for this class because it is really just an implementation detail for how IBFEMethod manages libMesh vectors - since all the IBFEMethod tests still pass this is fine.

File: ibtk/src/utilities/LibMeshSystemVectors.cpp [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+/////////////////////////////// STATIC ///////////////////////////////////////
+
+namespace
+{
+bool
```
Comment (boyceg) [inline ibtk/src/utilities/LibMeshSystemVectors.cpp]: Should this be `inline`?
```suggestion
inline bool
```

Comment (drwells) [reply ibtk/src/utilities/LibMeshSystemVectors.cpp]: If its in the same translation unit the compiler will consider it for inlining, so I don't think this matters.
